sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/SchemaUtils.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.meta.provider.datacatalog;
+
+import static org.apache.beam.sdk.schemas.Schema.toSchema;
+
+import com.google.cloud.datacatalog.ColumnSchema;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.extensions.sql.impl.utils.CalciteUtils;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.base.Strings;
+import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableMap;
+
+class SchemaUtils {
+
+  private static final Map<String, FieldType> FIELD_TYPES =
+      ImmutableMap.<String, FieldType>builder()
+          .put("TYPE_BOOL", FieldType.BOOLEAN)
+          .put("TYPE_BYTES", FieldType.BYTES)
+          .put("TYPE_DATE", FieldType.logicalType(new CalciteUtils.DateType()))
+          .put("TYPE_DATETIME", FieldType.DATETIME)
+          .put("TYPE_DOUBLE", FieldType.DOUBLE)
+          .put("TYPE_FLOAT", FieldType.DOUBLE)
+          .put("TYPE_INT32", FieldType.INT32)
+          .put("TYPE_INT64", FieldType.INT64)
+          .put("TYPE_STRING", FieldType.STRING)
+          .put("TYPE_TIME", FieldType.logicalType(new CalciteUtils.TimeType()))
+          .put("TYPE_TIMESTAMP", FieldType.DATETIME)
+          .put(
+              "TYPE_MAP<TYPE_STRING, TYPE_STRING>",
```

Comment:
Why MAP has only TYPE_STRING as Key and Value type? Is it because data catalog only allows STRING in MAP?

Reply:
No, this is how [`PubsubMessage` works](https://github.com/apache/beam/blob/b953645ed6db837d24284d7fe1fe091e7309f821/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubMessage.java#L32), same in [`CREATE TABLE`](https://beam.apache.org/documentation/dsls/sql/create-external-table/#syntax-2).

Reply:
Is there a reason that don't support MAP in a general way?